src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,165 @@
+package duke.parser;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.commands.AddCommand;
+import duke.commands.Command;
+import duke.commands.DeleteCommand;
+import duke.commands.ExitCommand;
+import duke.commands.FindCommand;
+import duke.commands.IncorrectCommand;
+import duke.commands.ListCommand;
+import duke.commands.MarkCommand;
+
+/**
+ * Represents the Parser Class.
+ * Responsible for parsing user input.
+ *
+ * @author Shishir
+ */
+public class Parser {
+
+    /**
+     * Returns a command based on user input.
+     * @param fullCommand String representation of user input.
+     * @return Command based on user input.
+     */
+    public static Command parse(String fullCommand) {
+
+        String[] split = fullCommand.split(" ", 2);
+        Command c = null;
+
+        switch(split[0]) {
+        case "bye":
+            c = validateCommand(split, true);
+            break;
+        case "list":
+            c = validateCommand(split, false);
+            break;
+        case "mark":
+        case "unmark":
+            c = validateIndex(split, true);
+            break;
+        case "todo":
+        case "deadline":
+        case "event":
+            c = validateTask(split);
+            break;
+        case "delete":
+            c = validateIndex(split, false);
+            break;
+        case "find":
+            c = validateFind(split);
+            break;
+        default:
+            c = new IncorrectCommand("I'm sorry, I couldn't understand that. Please try again!");
+        }
+        return c;
+    }
+
+    private static Command validateIndex(String[] split, boolean flag) {
+        // Check if mark is receiving any input or receiving extra input
+        if (split.length != 2 || split[1].isBlank()) {
+            return new IncorrectCommand("Please enter a valid mark command!");
+        }
+
+        // Check if mark is not receiving a number.
+        int index;
+        try {
+            index = Integer.parseInt(split[1]);
+        } catch (NumberFormatException exp) {
+            return new IncorrectCommand("I cannot mark a character! Please enter a number.");
+        }
+
+        // Check if index is greater than 0.
+        if (index <= 0) {
+            return new IncorrectCommand("Please enter a number greater than 0!");
+        }
+
+        return flag ? new MarkCommand(index, split[0]) : new DeleteCommand(index);
+    }
+
+    private static Command validateTask(String[] split) {
+
+        if (split.length == 1 || split[1].isBlank()) {
+            return new IncorrectCommand("Please enter a valid task.");
+        }
+
+        if (split[0].equals("deadline")) {
```

Comment:
I think addition of new task types might make this section complicated an error prone. (Imagine if you have a typo in the word "deadline", and it only appears here and not in the switch-case statement). Ditto this with the remaining sections.